### ISSUE TITLE
CO2.0 :  Child records appearing in related items and not redirecting [critical]

### ISSUE THREAD

**Issue by jamieu** [TheScienceMuseum/collectionsonline#1594]
Not sure what has changed as this was working earlier in the week and I don't _think_ the index has changed.

These SPH pages (both part records) are appearing in related items and throw an error.
http://localhost:8000/objects/co8428392/bbc-model-b-microcomputer-personal-computer
http://localhost:8000/objects/co64011/hard-disk-unit-for-apple-lisa-computer-1983-1991-computer-storage-device

- They seem to be part records (co8428392  = 1993-1086 Pt3 and co8428392  = 1995-632/1) which should redirect to the parent.
- They are also [appearing in the related items bloc at the bottom of the page](http://localhost:8000/objects/co8094437/next-computer-personal-computer), which is wrong, 'SPH part records' shouldn't appear there, it should use the same logic we use to suppress them in search results.

Once this is fixed maybe write a couple of tests to:

- [x] check a 'part' record redirects to the parent correctly.
- [x] check that a part isn't appearing its the related records.

```
[nodemon] clean exit - waiting for changes before restart
AssertError: Cannot set redirection mode without first setting a location
    at Response.permanent (/Users//projects/collectionsonline/node_modules/@hapi/hapi/lib/response.js:397:14)
    at handler (/Users//projects/collectionsonline/routes/object.js:59:47)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async exports.Manager.execute (/Users//projects/collectionsonline/node_modules/@hapi/hapi/lib/toolkit.js:60:28)
    at async internals.handler (/Users//projects/collectionsonline/node_modules/@hapi/hapi/lib/handler.js:46:20)
    at async exports.execute (/Users//projects/collectionsonline/node_modules/@hapi/hapi/lib/handler.js:31:20)
    at async Request._lifecycle (/Users//projects/collectionsonline/node_modules/@hapi/hapi/lib/request.js:370:32)
    at async Request._execute (/Users//projects/collectionsonline/node_modules/@hapi/hapi/lib/request.js:280:9)
Error: unavailable
    at handler (/Users//projects/collectionsonline/routes/object.js:67:23)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

```

**Comment by evancp87** [comment]
Hi Jamie, can you give an example of a record where these two are appearing as related items? Agree there should probably be more tests

**Comment by jamieu** [comment]
The NeXT computer, link below.
But also not working if I go direct to the part record , 

They are also [appearing in the related items bloc at the bottom of the page](http://localhost:8000/objects/co8094437/next-computer-personal-computer), which is wrong, 'SPH part records' shouldn't appear there, using the same logic we use to suppress them in search results

**Comment by evancp87** [comment]
> The NeXT computer, link below. But also not working if I go direct to the part record ,
> 
> They are also [appearing in the related items bloc at the bottom of the page](http://localhost:8000/objects/co8094437/next-computer-personal-computer), which is wrong, 'SPH part records' shouldn't appear there, it should use the same logic we use to suppress them in search results

Ok i'll have to take a proper look on Monday as i'm not getting errors for those records. Will look into suppressing the child records

**Comment by evancp87** [comment]
@jamieu I figured out what it is! It's because the parentSlug variable changed, this is where it was being used and the main reason i created that variable, so the redirect would take the user back to the page but with a slug for readability, rather than just the co number. I changed it in object.js to parent and the redirect works again. I'll commit on monday, as i will still need to supress child records from related items
